`dbt test` supports `--resource-type` / `--exclude-resource-type` CLI flags and related environment variables (#6126)
dbt-labs/docs.getdbt.com: website/docs/reference/global-configs/resource-type.md

sidebar: "resource type"
---

<VersionBlock lastVersion="1.8">

> The `--resource-type` and `--exclude-resource-type` flags include or exclude resource types from the `dbt build`, `dbt clone`, and `dbt list` commands.

Reviewer: non blocker but thought it might be helpful for 1.8 and lower users to know that newer versions/versionless supports dbt test
```suggestion
The `--resource-type` and `--exclude-resource-type` flags include or exclude resource types from the `dbt build`, `dbt clone`, and `dbt list` commands. Starting in Versionless and dbt v1.9, these flags are also supported in the `dbt test` command.
```

</VersionBlock>

<VersionBlock firstVersion="1.9">

> The `--resource-type` and `--exclude-resource-type` flags include or exclude resource types from the `dbt build`, `dbt test`, `dbt clone`, and `dbt list` commands.

Reviewer 2: I believe this resolves the comment in https://github.com/dbt-labs/docs.getdbt.com/pull/6126#discussion_r1781351928.

Namely, as-of v1.9, there are 4 different subcommands that support these flags, and this sentence lists all of them: `build`, `test`, `clone`, and `list`.

</VersionBlock>

> <VersionBlock lastVersion="1.8">

Reviewer 3: Should we keep the firstversion rule in since we still display v1.6 and v1.7?
```suggestion
<VersionBlock firstVersion="1.8" lastVersion="1.8">
```

</VersionBlock>

> <VersionBlock firstVersion="1.8" lastVersion="1.8">

Reviewer 2: I think this listing can remain as-is, so restoring to original

```suggestion
<VersionBlock firstVersion="1.8">
```

</VersionBlock>

<VersionBlock firstVersion="1.9">

- [`analysis`](/docs/build/analyses)
- [`exposure`](/docs/build/exposures)
- [`metric`](/docs/build/build-metrics-intro)
- [`model`](/docs/build/models)
- [`saved_query`](/docs/build/saved-queries)
- [`seed`](/docs/build/seeds)
- [`semantic_model`](/docs/build/semantic-models)
- [`snapshot`](/docs/build/snapshots)
- [`source`](/docs/build/sources)
- [`test`](/docs/build/data-tests)

> - [`unit_test`](/docs/build/unit-tests)

Reviewer 4: This list looks to be the same in 1.9 and earlier versions, why is it in the version block?

> dbt `test` now enables passing values such as `unit_tests`, `tests`, and more in the `--resource-type` and `--exclude-resource-type` for CLI flags (or in the environment variables `DBT_RESOURCE_TYPES` and `DBT_EXCLUDE_RESOURCE_TYPES)`.

Reviewer 4: I don't think we need callouts like "_now_" - we should:
- have a callout that this is available in versionless and starting in dbt Core `v1.9`
- then document the feature

> dbt `test` now enables passing values such as `unit_tests`, `tests` and more so in the `--resource-type` and `--exclude-resource-type` for CLI flags (or in the environment variables `DBT_RESOURCE_TYPES` and `DBT_EXCLUDE_RESOURCE_TYPES)`.

Reviewer 3: ```suggestion
dbt `test` now enables passing values such as `unit_tests`, `tests`, and more in the `--resource-type` and `--exclude-resource-type` for CLI flags (or in the environment variables `DBT_RESOURCE_TYPES` and `DBT_EXCLUDE_RESOURCE_TYPES)`.
```

> Here are some examples of how dbt `test` enables passing values:

Reviewer 3: "Here" can be a little vague, so we generally suggest specific language such as "the following..." or "the previous..."
```suggestion
The following are some examples of how dbt `test` enables passing values:
```

> The following are some examples of how dbt `test` enables passing values:

Reviewer 4: We don't need a separate section for using this config with `dbt test`, we already have examples below on using it with `dbt build`. These can be the same section.

<Expandable alt_header="CLI flag">

```
dbt `test` --resource-type unit_test
dbt `test` --exclude-resource-type unit_test

```

</Expandable>

<Expandable alt_header="Environment variables">

```

`DBT_RESOURCE_TYPES`=`unit_test` dbt `test`
`DBT_EXCLUDE_RESOURCE_TYPES`=`unit_test` dbt `test`

```

</Expandable>

</VersionBlock>

## Example

> Instead of targeting specific resources, use the `--resource-flag` or `--exclude-resource-type` flags to target all resources of a certain type: `dbt build --resource-type RESOURCE_TYPE` replacing `RESOURCE_TYPE` with the resource type you want to include.

Reviewer 4: ```suggestion
Instead of targeting specific resources, use the `--resource-flag` or `--exclude-resource-type` flags with the `build` or `test` command to target all resources of a certain type: `dbt build --resource-type RESOURCE_TYPE` replacing `RESOURCE_TYPE` with the resource type you want to include.
```

I think this is all we need?

Reviewer 2: See https://github.com/dbt-labs/docs.getdbt.com/pull/6126/files#r1782100963 which I believe accomplishes this same thing.

</VersionBlock>

> <VersionBlock firstVersion="1.9">
> 
> - [`analysis`](/docs/build/analyses)
> - [`exposure`](/docs/build/exposures)
> - [`metric`](/docs/build/build-metrics-intro)
> - [`model`](/docs/build/models)
> - [`saved_query`](/docs/build/saved-queries)
> - [`seed`](/docs/build/seeds)
> - [`semantic_model`](/docs/build/semantic-models)
> - [`snapshot`](/docs/build/snapshots)
> - [`source`](/docs/build/sources)
> - [`test`](/docs/build/data-tests)
> - [`unit_test`](/docs/build/unit-tests)
> 
> dbt `test` now enables passing values such as `unit_tests`, `tests`, and more in the `--resource-type` and `--exclude-resource-type` for CLI flags (or in the environment variables `DBT_RESOURCE_TYPES` and `DBT_EXCLUDE_RESOURCE_TYPES)`.
> 
> The following are some examples of how dbt `test` enables passing values:
> 
> <Expandable alt_header="CLI flag">
> 
> ```
> dbt `test` --resource-type unit_test
> dbt `test` --exclude-resource-type unit_test
> 
> ```
> 
> </Expandable>
> 
> <Expandable alt_header="Environment variables">
> 
> ```
> 
> `DBT_RESOURCE_TYPES`=`unit_test` dbt `test`
> `DBT_EXCLUDE_RESOURCE_TYPES`=`unit_test` dbt `test`
> 
> ```
> 
> </Expandable>
> 
> </VersionBlock>
> 

Reviewer 2: ```suggestion
```

    </File>

</VersionBlock>

<VersionBlock firstVersion="1.9">

> -  In this example, use the following command to exclude _all_ unit tests when running tests. Note that the `--exclude-resource-type` flag is only available in dbt version 1.8 and higher:

Reviewer 4: 1.8 or 1.9?

Reviewer 2: Oops, missed that during copy-paste -- good catch!

Removing the "dbt version x.y and higher piece since this is within a `<Version>` tag and will only show when Versionless or v1.9+ is chosen. 

```suggestion
-  In this example, use the following command to exclude _all_ unit tests when running tests:
```
